opcua/client/ua_client.py

```diff
@@ -400,23 +400,32 @@ def publish(self, acks=None):
             acks = []
         request = ua.PublishRequest()
         request.Parameters.SubscriptionAcknowledgements = acks
-        self._uasocket.send_request(request, self._call_publish_callback, timeout=int(9e8))  # timeout could be set to 0 but some servers to not support it
+        # timeout could be set to 0 but some servers to not support its
+        self._uasocket.send_request(request, self._call_publish_callback, timeout=int(9e8)) # 250 days
 
     def _call_publish_callback(self, future):
         self.logger.info("call_publish_callback")
         data = future.result()
-        self._uasocket.check_answer(data, "ServiceFault received from server while waiting for publish response")
+
+        try:
+            self._uasocket.check_answer(data, "while waiting for publish response")
+        except BadTimeout: # Spec Part 4, 7.28
+            self.publish()
+
         try:
             response = ua.PublishResponse.from_binary(data)
             self.logger.debug(response)
         except Exception:
             self.logger.exception("Error parsing notificatipn from server")
             self.publish([]) #send publish request ot server so he does stop sending notifications
```

Comment:
Does anyone know in which cases this happens? It seems to me that catching `BadTimeout` has we now do above is the only case re-sending a publish request is appropriate.